bin/tasks/gitInfo.js

```diff
@@ -52,31 +60,61 @@ export const setGitInfo = async (ctx, task) => {
     // The SkipBuildMutation ensures the commit is tagged properly.
     if (await ctx.client.runQuery(TesterSkipBuildMutation, { commit })) {
       ctx.skip = true;
-      return transitionTo(skippedForCommit, true)(ctx, task);
+      transitionTo(skippedForCommit, true)(ctx, task);
+      return;
     }
     throw new Error(skipFailed(ctx).output);
   }
 
-  const baselineCommits = await getBaselineCommits(ctx, {
-    branch,
+  const parentCommits = await getParentCommits(ctx, {
     ignoreLastBuildOnBranch: matchesBranch(ctx.options.ignoreLastBuildOnBranch),
   });
-  ctx.git.baselineCommits = baselineCommits;
-  ctx.log.debug(`Found baselineCommits: ${baselineCommits}`);
+  ctx.git.parentCommits = parentCommits;
+  ctx.log.debug(`Found parentCommits: ${parentCommits.join(', ')}`);
 
-  // If the sole baseline is the most recent ancestor, then this is likely a rebuild (rerun of CI job).
+  // If we're running against the same commit as the sole parent, then this is likely a rebuild (rerun of CI job).
   // If the MRA is all green, there's no need to rerun the build, we just want the CLI to exit 0 so the CI job succeeds.
   // This is especially relevant for (unlinked) projects that don't use --exit-zero-on-changes.
   // There's no need for a SkipBuildMutation because we don't have to tag the commit again.
-  if (baselineCommits.length === 1 && baselineCommits[0] === commit) {
+  if (parentCommits.length === 1 && parentCommits[0] === commit) {
     const mostRecentAncestor = await ctx.client.runQuery(TesterLastBuildQuery, { commit, branch });
     if (mostRecentAncestor && ['PASSED', 'ACCEPTED'].includes(mostRecentAncestor.status)) {
       ctx.skip = true;
-      return transitionTo(skippedRebuild, true)(ctx, task);
+      transitionTo(skippedRebuild, true)(ctx, task);
+      return;
+    }
+  }
+
+  // Retrieve a list of changed file paths since the actual baseline commit(s), which will be used
+  // to determine affected story files later.
+  // In the unlikely scenario that this list is empty (and not a rebuild), we can skip the build
+  // since we know for certain it wouldn't have any effect. We do want to tag the commit.
+  if (parentCommits.length && matchesBranch(ctx.options.onlyChanged)) {
+    const baselineCommits = await getBaselineCommits(ctx, { branch, parentCommits });
+    ctx.log.debug(`Found baselineCommits: ${baselineCommits.join(', ')}`);
+
+    try {
+      const results = await Promise.all(baselineCommits.map((c) => getChangedFiles(c)));
+      ctx.git.changedFiles = [...new Set(results.flat())].map((f) => `./${f}`);
+      ctx.log.debug(`Found changedFiles:\n${ctx.git.changedFiles.map((f) => `  ${f}`).join('\n')}`);
```

Comment:
```suggestion
      // Find the union of all the changes since each baselineCommit.
      // If a file has been modified since one side of a merge but not the other, we don't know which
      // side contains the baseline, so we need to be conservative and assume it has changed since the baseline.
      const results = await Promise.all(baselineCommits.map((c) => getChangedFiles(c)));
      ctx.git.changedFiles = [...new Set(results.flat())].map((f) => `./${f}`);
      ctx.log.debug(`Found changedFiles:\n${ctx.git.changedFiles.map((f) => `  ${f}`).join('\n')}`);
```

Reply:
Technically, we could determine the changes for each commit individually. I don't think that's worthwhile though.
I've now implemented this such that we select the most recently committed baseline build as "the baseline" to determine the build status from. Or should we look at the build statuses instead?